dproc/dproc.go

```diff
@@ -57,22 +127,22 @@ func GetDataprocRequestParams(sourceProfile profiles.SourceProfile, targetProfil
 	if len(sourceProfile.Config.ShardConfigurationDataproc.DataprocConfig.Port) > 1 {
 		port = sourceProfile.Config.ShardConfigurationDataproc.DataprocConfig.Port
 	}
+	spTableName := conv.SpSchema[spTableID].Name
 
 	jdbcParams := map[string]string{}
-	// TODO: replace hardcoded `synth_id` with a variable/constant
 	if sourceProfile.Driver == constants.MYSQL {
 		jdbcParams["url"] = fmt.Sprintf("jdbc:mysql://%s:%s/%s?user=%s&password=%s", host, port, srcDb, user, pwd)
 		jdbcParams["driver"] = "com.mysql.jdbc.Driver"
-		if primaryKeys == "" {
-			jdbcParams["sql"] = fmt.Sprintf("select uuid() as synth_id, t.* from %s.%s as t", srcSchema, srcTable)
-			primaryKeys = "synth_id"
-		} else {
-			jdbcParams["sql"] = fmt.Sprintf("select * from %s.%s", srcSchema, srcTable)
-		}
 		jdbcParams["jar"] = "gs://dataproc-templates/jars/mysql-connector-java.jar"
 	} else {
 		return DataprocRequestParams{}, fmt.Errorf("dataproc migration for source %s not supported", sourceProfile.Driver)
 	}
+	jdbcParams["sql"] = getJdbcSql(conv, sourceProfile.Driver, spTableID, primaryKeys)
+
+	outputPrimaryKeys := primaryKeys
```

Comment:
why is this required in the template? What happens if user edits the primary key ?

Reply:
Currently, 
* In case of schema+data migration via template: Dataproc template uses primary key parameter to define primary key in target spanner database when doing 
* **[OUR CASE]** In case of data migration only (schema created already before triggering template) this parameter is not used (it even works if given any random value to the parameter)

BUT, In both cases, this parameter is set as required parameter, (throws error: `ValidationUtil: Violation: primaryKey jdbctospanner.output.primaryKey; must not be empty` if not given)

@enclona-google, Please add if i missed anything

Reply:
Looking at current dataproc template code, it's not a very simple change to identify migration type in the validation phase, I suggest until that change is done, We keep on passing the actual primary key as we're doing now in harbourbridge.

Reply:
Added comment explaining same
Commit: https://github.com/GoogleCloudPlatform/spanner-migration-tool/pull/610/commits/956615ac6f5c5c5ad57ffc972404a3702092a3e2